solr/solrj/src/java/org/apache/solr/client/solrj/impl/CloudHttp1SolrClient.java

```diff
@@ -0,0 +1,351 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.solr.client.solrj.impl;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import org.apache.http.NoHttpResponseException;
+import org.apache.http.client.HttpClient;
+import org.apache.http.conn.ConnectTimeoutException;
+import org.apache.solr.client.solrj.request.UpdateRequest;
+import org.apache.solr.common.SolrException;
+import org.apache.solr.common.cloud.DocCollection;
+import org.apache.solr.common.cloud.DocRouter;
+import org.apache.solr.common.params.ModifiableSolrParams;
+import org.apache.solr.common.util.NamedList;
+
+/**
+ * SolrJ client class to communicate with SolrCloud. Instances of this class communicate with
+ * Zookeeper to discover Solr endpoints for SolrCloud collections, and then use the {@link
+ * LBHttpSolrClient} to issue requests.
+ *
+ * @deprecated Please use {@link CloudHttp2SolrClient}
+ */
+@Deprecated(since = "9.0")
+public class CloudHttp1SolrClient extends CloudSolrClient {
```

Comment:
As you also noted, *Http1* here is misleading, since CloudHttp2SolrClient is *capable* of http2, but also of http1.

Have you considered `CloudApacheHttpSolrClient` or similar to point to the fact that apache-http is used? Or if we want to hint that it is using `LBHttpSolrClient` and `HttpSolrClient` under the hood, the name should be simply `CloudHttpSolrClient` for a strong name-similarity?

Reply:
Indeed I've thought of this but stopped short because we hadn't discussed it yet.  Indeed, I would prefer a name like CloudApacheHttpSolrClient or CloudLegacySolrClient.  I'll raise this in the list for visibility.